vpn/src/main/java/com/duckduckgo/mobile/android/vpn/feature/removal/VpnFeatureRemover.kt

```diff
@@ -0,0 +1,117 @@
+import com.squareup.anvil.annotations.ContributesBinding
+import com.squareup.anvil.annotations.ContributesMultibinding
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+import javax.inject.Provider
+
+interface VpnFeatureRemover {
+    fun manuallyRemoveFeature()
+    fun scheduledRemoveFeature()
+}
+
+@ContributesBinding(scope = AppScope::class, boundType = VpnFeatureRemover::class)
+@ContributesMultibinding(scope = AppScope::class, boundType = WorkerInjectorPlugin::class)
+class DefaultVpnFeatureRemover @Inject constructor(
+    private val deviceShieldOnboarding: DeviceShieldOnboardingStore,
+    private val notificationManager: NotificationManagerCompat,
+    private val vpnDatabase: VpnDatabase,
+    // we use the Provider to avoid a cycle dependency
+    private val workManagerProvider: Provider<WorkManager>,
+    @AppCoroutineScope private val appCoroutineScope: CoroutineScope,
+    private val dispatcherProvider: DispatcherProvider
+) : VpnFeatureRemover, WorkerInjectorPlugin {
+
+    // Disabling reminder notifications
+    // Disable daily / weekly notifications
+    // Remove CTA in New Tab
+    // Remove all app trackers blocked
+    // Remove app shortcut
+    // If manual removal
+    // Force users to complete AppTP Onboarding
+
+    override fun inject(worker: ListenableWorker): Boolean {
+        if (worker is VpnFeatureRemoverWorker) {
+            worker.vpnFeatureRemover = this
+            return true
+        }
+
+        return false
+    }
+
+    override fun manuallyRemoveFeature() {
+        appCoroutineScope.launch(dispatcherProvider.io()) {
+            disableNotifications()
+            disableNotificationReminders()
+            removeNotificationChannels()
+            deleteAllVpnTrackers()
+            disableFeature()
+        }
+    }
+
+    private fun disableNotificationReminders() {
+        val workManager = workManagerProvider.get()
+        workManager.cancelAllWorkByTag(VpnReminderNotificationWorker.WORKER_VPN_REMINDER_UNDESIRED_TAG)
+        workManager.cancelAllWorkByTag(VpnReminderNotificationWorker.WORKER_VPN_REMINDER_DAILY_TAG)
+    }
+
+    override fun scheduledRemoveFeature() {
+        manuallyRemoveFeature()
+        resetAppTPOnboarding()
+    }
+
+    private fun disableNotifications() {
+        notificationManager.cancel(TrackerBlockingVpnService.VPN_REMINDER_NOTIFICATION_ID)
+        notificationManager.cancel(VPN_DAILY_NOTIFICATION_ID)
+        notificationManager.cancel(VPN_WEEKLY_NOTIFICATION_ID)
+    }
+
+    private fun removeNotificationChannels() {
+        notificationManager.deleteNotificationChannel(AndroidDeviceShieldAlertNotificationBuilder.VPN_ALERTS_CHANNEL_ID)
+    }
+
+    private fun resetAppTPOnboarding() {
+        deviceShieldOnboarding.onboardingDidNotShow()
+    }
+
+    private fun deleteAllVpnTrackers() {
+        vpnDatabase.vpnTrackerDao().deleteAllTrackers()
```

Comment:
Why not deleting VPN related tables?
```suggestion
        vpnDatabase.clearAllTables()
```